command/agent/consul/connect.go

```diff
@@ -64,6 +66,10 @@ func newConnectGateway(serviceName string, connect *structs.ConsulConnect) *api.
 			envoyConfig["envoy_gateway_bind_tagged_addresses"] = true
 		}
 
+		if proxy.EnvoyDNSDiscoveryType != "" {
```

Comment:
It was left out for ingress because it only has meaning for terminating and mesh gateways
https://www.consul.io/docs/connect/proxies/envoy#envoy_dns_discovery_type

Reply:
Gotcha 👍